common/src/main/java/com/microsoft/identity/common/internal/cache/registry/DefaultBrokerApplicationRegistry.java

```diff
@@ -0,0 +1,212 @@
+package com.microsoft.identity.common.internal.cache.registry;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+
+import androidx.annotation.NonNull;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import com.microsoft.identity.common.internal.logging.Logger;
+
+import java.lang.reflect.Type;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class DefaultBrokerApplicationRegistry implements IBrokerApplicationRegistry {
+
+    private static final String TAG = DefaultBrokerApplicationRegistry.class.getSimpleName();
+
+    private static final String DEFAULT_APP_REGISTRY_CACHE_NAME = "com.microsoft.identity.app-registry";
+    private static final String KEY_APP_REGISTRY = "app-registry";
+    private static final String EMPTY_ARRAY = "[]";
+
+    private final SharedPreferences mSharedPrefs;
+
+    private final Gson mGson = new Gson();
+
+    public DefaultBrokerApplicationRegistry(@NonNull final Context context) {
+        mSharedPrefs = context.getSharedPreferences(
+                DEFAULT_APP_REGISTRY_CACHE_NAME,
+                Context.MODE_PRIVATE
+        );
+    }
+
+    @Override
+    public BrokerApplicationRegistryData getMetadata(@NonNull final String clientId,
+                                                     @NonNull final String environment,
```

Comment:
Let's make `environment` `Nullable` as discussed since getAccounts() wouldn't have environment information

Reply:
Done. Resolved in 5720d395cdbb990cb032089d780d206ae554e575